Sources/Native/NativeMockServerWrapper.swift

```diff
@@ -2,102 +2,206 @@ import Foundation
 import SwiftyJSON
 import BrightFutures
 
-class NativeMockServerWrapper: MockServer {
-  var port: Int32 = -1
-  var pactDir: String = ""
-
-  public init() {
-    pactDir = defaultPactDir()
-    port = randomPort()
-  }
-
-  func defaultPactDir() -> String {
-    if let dir = ProcessInfo.processInfo.environment["pact_dir"] {
-      return dir
+#if SWIFT_PACKAGE
+    import SwiftPactServer
+#endif
+
+public class NativeMockServerWrapper: MockServer {
+    private lazy var port: Int32 = {
+        return findUnusedPort()
+    }()
+    private let pactDir: String = ProcessInfo.processInfo.environment["pact_dir"] ?? "/tmp/pacts"
+    private let shouldWritePacts: Bool
+
+    public init(shouldWritePacts: Bool = false) {
+        self.shouldWritePacts = shouldWritePacts
+    }
+
+    private func findUnusedPort() -> Int32 {
+        var port = randomPort
+        var (unused, description) = checkTcpPortForListen(port: port)
+        while !unused {
+            debugPrint(description)
+            port = randomPort
+            (unused, description) = checkTcpPortForListen(port: port)
+        }
+        debugPrint(description)
+        return Int32(port)
+    }
+
+    private var randomPort: in_port_t {
+        return in_port_t(arc4random_uniform(2000) + 4000)
+    }
+
+    public func getBaseUrl() -> String {
+        return "http://localhost:\(port)"
+    }
+
+    public func setup(_ pact: Pact) -> Future<String, PactError> {
+        return Future { complete in
+            do {
+                let payload = pact.payload()
+                let jsonData = try JSONSerialization.data(withJSONObject: payload)
+                let jsonString = String(bytes: jsonData, encoding: String.Encoding.utf8)
+
+                // iOS json generation adds extra backslashes to "application/json" --> "application\\/json"
+                // causing the MockServer to fail to parse the file.
+                let sanitizedString = jsonString!.replacingOccurrences(of: "\\/", with: "/")
```

Comment:
This sanitisation is occurring in a few places, could pull it out into a reusable method or a String extension.

Reply:
TODO: 
always a good idea